lib/app/sequence_base.rb

```diff
@@ -524,7 +524,8 @@ def validate_search_reply(klass, reply, search_params)
       def validate_read_reply(resource, klass)
         class_name = klass.name.demodulize
         assert !resource.nil?, "No #{class_name} resources available from search."
-        if resource.is_a? FHIR::DSTU2::Reference
+        resource_is_reference = resource.is_a? FHIR::DSTU2::Reference
```

Comment:
Does this line only work for DSTU2 servers?  The name of the method doesn't seem to indicate it is only for DSTU2.

Reply:
Could you use `resource.is_a? versioned_resource_class('Reference')`

Reply:
Removed logic for reference altogether since it's not being used anywhere.